sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -193,10 +196,13 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
       if (ds.isInstanceOf[ReadSupport] || ds.isInstanceOf[ReadSupportWithSchema]) {
         val sessionOptions = DataSourceV2Utils.extractSessionConfigs(
           ds = ds, conf = sparkSession.sessionState.conf)
+        val pathsOption = {
+          val objectMapper = new ObjectMapper()
+          DataSourceOptions.PATHS_KEY -> objectMapper.writeValueAsString(paths.toArray)
+        }
         Dataset.ofRows(sparkSession, DataSourceV2Relation.create(
-          ds, extraOptions.toMap ++ sessionOptions,
+          ds, extraOptions.toMap ++ sessionOptions + pathsOption,
```

Comment:
issue an exception when extraOptions("path") is not empty?

Reply:
Basically we may have duplicated entries in session configs and `DataFrameReader/Writer` options, not only path. The rule is, `DataFrameReader/Writer` options should overwrite session configs.

cc @jiangxb1987 can you submit a PR to explicitly document it in `SessionConfigSupport`?